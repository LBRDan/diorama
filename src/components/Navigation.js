import React from 'react';
import PropTypes from 'prop-types';

import style from './Navigation.css';

const Navigation = ({ onPreviousSlide, onNextSlide }) => (
  <nav className={`${style.navigation} diorama-navigation`}>
    <ul className={style.list}>
      <li className={style.item}>
        <button onClick={onPreviousSlide} className={`${style.previous} diorama-previous`}>
          Previous
        </button>
      </li>
      <li className={style.item}>
        <button onClick={onNextSlide} className={`${style.next} diorama-next`}>
          Next
        </button>
      </li>
    </ul>
  </nav>
);

Navigation.propTypes = {
  onPreviousSlide: PropTypes.func.isRequired,
  onNextSlide: PropTypes.func.isRequired,
};

export default Navigation;
