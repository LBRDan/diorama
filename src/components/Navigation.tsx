import React from "react";
import PropTypes from "prop-types";

import styles from "./Navigation.module.css";

function Navigation({ onPreviousSlide, onNextSlide, className, style }) {
  return <nav
		className={`${styles.navigation} diorama-navigation ${className}`}
		style={style}
	>
		<ul className={styles.list}>
			<li className={styles.item}>
				<button
					onClick={onPreviousSlide}
					className={`${styles.previous} diorama-previous`}
					type="button"
				>
					Previous
				</button>
			</li>
			<li className={styles.item}>
				<button
					onClick={onNextSlide}
					className={`${styles.next} diorama-next`}
					type="button"
				>
					Next
				</button>
			</li>
		</ul>
	</nav>
}

Navigation.propTypes = {
	className: PropTypes.string,
	onPreviousSlide: PropTypes.func.isRequired,
	onNextSlide: PropTypes.func.isRequired,
	style: PropTypes.shape({}),
};

Navigation.defaultProps = {
	className: "",
	style: {},
};

export default Navigation;
