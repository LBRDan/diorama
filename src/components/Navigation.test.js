import React from 'react';
import renderer from 'react-test-renderer';
import Navigation from './Navigation';

it('renders correctly', () => {
  const tree = renderer
    .create(<Navigation onPreviousSlide={() => true} onNextSlide={() => true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
