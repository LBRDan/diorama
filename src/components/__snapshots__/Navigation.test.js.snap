// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navigation diorama-navigation"
>
  <ul
    className="list"
  >
    <li
      className="item"
    >
      <button
        className="previous diorama-previous"
        onClick={[Function]}
      >
        Previous
      </button>
    </li>
    <li
      className="item"
    >
      <button
        className="next diorama-next"
        onClick={[Function]}
      >
        Next
      </button>
    </li>
  </ul>
</nav>
`;
